### ISSUE TITLE
[tmva][sofie] Conv2D with data_format=channels_first and padding=same gives wrong output shape

### ISSUE THREAD

**Issue by markknoffler** [root-project/root#21661]
### Check duplicate issues.

- [x] Checked for duplicates

### Description

When parsing a Keras Conv2D layer with `data_format='channels_first'` and `padding='same'`, the SOFIE Keras parser computes padding using wrong indices. It always uses indices 1 and 2 for height and width, which is correct for channels_last (NHWC) but wrong for channels_first (NCHW). For channels_first the shape is (N, C, H, W) so H is at index 2 and W is at index 3. Because of this the output tensor shape does not match Keras and the inference results are wrong.

<img width="1468" height="491" alt="Image" src="https://github.com/user-attachments/assets/5e5a52b1-ec95-4258-bc87-fce609c873d1" />

### Reproducer

1. Source ROOT: `source /path/to/root-build/bin/thisroot.sh`
2. Run this Python script:

```python
import os
import tempfile
import numpy as np
import ROOT
from keras import layers, models

model = models.Sequential([
    layers.Input(shape=(2, 5, 5)),
    layers.Conv2D(4, (3, 3), padding='same', strides=(2, 2),
                 data_format='channels_first', activation='relu')
])
model.compile(optimizer='adam', loss='mean_squared_error')
x = np.random.rand(2, 2, 5, 5).astype('float32')
y = np.random.rand(2, 4, 3, 3).astype('float32')
model.fit(x, y, epochs=1, verbose=0)

with tempfile.TemporaryDirectory() as tmpdir:
    model_path = os.path.join(tmpdir, "conv2d_channels_first_same.keras")
    model.save(model_path)
    rmodel = ROOT.TMVA.Experimental.SOFIE.PyKeras.Parse(model_path, batch_size=1)
    hxx_path = os.path.join(tmpdir, "conv2d_channels_first_same.hxx")
    rmodel.Generate()
    rmodel.OutputGenerated(hxx_path)
    ROOT.gInterpreter.Declare('#include "' + hxx_path + '"')
    sess = getattr(ROOT, "TMVA_SOFIE_conv2d_channels_first_same").Session(hxx_path.replace(".hxx", ".dat"))
    km = models.load_model(model_path)
    inp = np.ones((1, 2, 5, 5), dtype='float32')
    sofie_out = np.asarray(sess.infer(inp)).flatten()
    keras_out = np.asarray(km(inp)).flatten()
    print("SOFIE size:", sofie_out.size)
    print("Keras size:", keras_out.size)
```
3. Without fix: SOFIE outputs 24 elements, Keras outputs 36. They should be same.

4. Run: `python script.py` (with ROOT and Keras in path)

<img width="1468" height="491" alt="Image" src="https://github.com/user-attachments/assets/8627a919-8398-4094-9d1d-7b1595e0af7e" />

### ROOT version

ROOT 6.39.01 

### Installation method

Built from source

### Operating system

Linux (Ubuntu 24.04)

### Additional context

Fix is in `bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_sofie/_parser/_keras/layers/conv.py`. For same padding with channels_first, use `fInputShape[2]` and `fInputShape[3]` for inputHeight and inputWidth instead of `fInputShape[1]` and `fInputShape[2]`.
